churaverse-plugins-client/src/gamePlugin/domain/coreGamePlugin.ts

```diff
@@ -0,0 +1,93 @@
+import { GameAbortEvent } from '../event/gameAbortEvent'
+import { GameEndEvent } from '../event/gameEndEvent'
+import { GameStartEvent } from '../event/gameStartEvent'
+import { PriorGameDataEvent } from '../event/priorGameDataEvent'
+import { GameIds } from '../interface/gameIds'
+import { GamePluginStore } from '../store/defGamePluginStore'
+import { BasicGamePlugin } from './basicGamePlugin'
+
+/**
+ * BasicGamePluginを拡張したCoreなゲーム抽象クラス
+ */
+export abstract class CoreGamePlugin extends BasicGamePlugin {
+  protected abstract gameId: GameIds
+  protected abstract gameName: string
+  private _isActive: boolean = false
+  private _gameOwnerId?: string
+  private _participantIds: string[] = []
+  private _isOwnPlayerMidwayParticipant: boolean = false
+  protected gamePluginStore!: GamePluginStore
+
+  public get isActive(): boolean {
+    return this._isActive
+  }
+
+  public get gameOwnerId(): string | undefined {
+    return this._gameOwnerId
+  }
+
+  public get participantIds(): string[] {
+    return this._participantIds
+  }
+
+  public get isOwnPlayerMidwayParticipant(): boolean {
+    return this._isOwnPlayerMidwayParticipant
+  }
+
+  public listenEvent(): void {
+    this.bus.subscribeEvent('gameStart', this.gameStart.bind(this))
+    this.bus.subscribeEvent('priorGameData', this.priorGameData.bind(this))
+  }
+
+  protected subscribeGameEvent(): void {
+    this.bus.subscribeEvent('gameAbort', this.gameAbort)
+    this.bus.subscribeEvent('gameEnd', this.gameEnd)
```

Comment:
BasicGamePluginの関数をoverrideしているので、handleGameTermination等々がsubscribeEventされない気がしているんですが確認してもらって良いですか？

super.subscribeGameEvent()でうまく動けばDRYになって良さそうですが、thisのバインドの都合上うまく動くか怪しいのでその辺りの挙動も注意して確認して欲しいです
```suggestion
    super.subscribeGameEvent()
    this.bus.subscribeEvent('gameAbort', this.gameAbort)
    this.bus.subscribeEvent('gameEnd', this.gameEnd)
```